tests/plugins/dut_monitor/pytest_dut_monitor.py

```diff
@@ -0,0 +1,358 @@
+
+    def exec_command(self, cmd, timeout=None):
+        """
+        @summary: Execute a command on the DUT and track possible connectivity issues. A new Channel is opened and the requested command is executed
+        """
+        try:
+            return self.ssh.exec_command(cmd, timeout=timeout, get_pty=True)
+        except Exception as err:
+            logger.warning("Broken connection - {}".format(repr(err)))
+            logger.warning("Skip command {}".format(cmd))
+            return (None, None, None)
+
+    def start(self):
+        """
+        @summary: Start HW resources monitoring on the DUT. Write obtained values to the following files on the DUT: DUT_CPU_LOG, DUT_RAM_LOG, DUT_HDD_LOG
+        """
+        self.running = True
+        self._upload_to_dut()
+        logger.debug("Start HW resuorces monitoring on the DUT...")
+
+        self.run_channel = self.ssh.get_transport().open_session()
+        self.run_channel.get_pty()
+        self.run_channel.settimeout(5)
+        # Start monitoring on DUT
+        self.run_channel.exec_command("python {} --start".format(DUT_MONITOR))
+        # Ensure monitoring started
+        output = self.run_channel.recv(1024)
+        if not "Started resources monitoring ..." in output:
+            raise Exception("Failed to start monitoring on DUT: {}".format(output))
+
+    def stop(self):
+        """
+        @summary: Close this SSHClient and its underlying Transport
+        """
+        self.running = False
+        logger.debug("Stop resources monitoring on the DUT...")
+        if not self.run_channel.closed:
+            self.run_channel.close()
+
+    def read_yml(self, file_pointer):
+        """
+        @summary: Read yaml file content. Convert it to the ordered data.
+        @return: OrderedDict with sorted keys by timestamp, or empty dict for empty file.
+        """
+        with file_pointer as fp:
+            measurements = yaml.safe_load("".join(fp))
+        if measurements is None:
+            return {}
+        # Sort json data to process logs chronologically 
+        keys = measurements.keys()
+        keys.sort()
+        key_value_pairs = [(item, measurements[item]) for item in keys]
+        return OrderedDict(key_value_pairs)
+
+    def get_log_files(self):
+        """
+        @summary: Fetch monitoring logs from device, parse, convert to dictionary with sorted order.
+        @return: Dictionary with keys "cpu", "ram", "hdd", values contains appropriate measurements made on DUT.
+        """
+        logger.debug("Downloading file from the DUT...")
+        cpu_log_fp = self.ssh.open_sftp().file(DUT_CPU_LOG)
+        ram_log_fp = self.ssh.open_sftp().file(DUT_RAM_LOG)
+        hdd_log_fp = self.ssh.open_sftp().file(DUT_HDD_LOG)
+
+        cpu_meas = self.read_yml(cpu_log_fp)
+        ram_meas = self.read_yml(ram_log_fp)
+        hdd_meas = self.read_yml(hdd_log_fp)
+        return {"cpu": cpu_meas, "ram": ram_meas, "hdd": hdd_meas}
```

Comment:
It would be nice if there is a README.md document explaining how to use it and how it works.

Reply:
There is HLD:
https://github.com/Azure/SONiC/blob/master/doc/DUT_monitor_HLD.md

Or you mean adding some more specific explanation?

Reply:
The HLD document is good. Maybe add a README.md and refer to the HLD document in it?

Reply:
Ok, will add README.md into "dut_monitor" folder.